### PR TITLE
SAL-1478: Implement the loaded? functionality

### PR DESCRIPTION
**Pull request by josemarialopez**
# 🤔 What issue does this fix?

### Context

In order to optimise the performance of the services at Glue that are manipulating information at Salesforce through the ActiveForce gem, we would require an equivalent version to the loaded? method available for ActiveRecord associations. The objective of this card is to create that method.

### Acceptance criteria

* A loaded? method has been created for ActiveForce::ActiveQuery instances so we can determine if an association has already bee loaded on memory or not.

# 🍂 Before You Submit:

- [x] Did you write tests?
- [ ] Did you write [documentation on Notion](https://www.notion.so/30da1978bf5e4370960ea2e7dd4c6940?v=d86eb275909341bc9deeb704e6cd876a)? (N/A)
- [x] Did you lint?
- [x] Did you run tests?
- [x] Did you link this PR in Trello?
- [ ] Did you update any env changes in the `.env.example` and on [Notion](https://www.notion.so/1e85593da10744e09affb7642540c83f?v=88550674a18b43789f9acdb7bf518676)? (N/A)

# 🛷 Deployment Considerations

- [ ] Data Migrations (N/A)
- [ ] Schema Migrations (N/A)
- [ ] Dependencies (N/A)

# 🧪 How to Test

* We have implemented some tests for this new logic. We can rely on them
* It's possible to test it from the flue repository pointing to the local version of the active force gem and taking any `ActiveForce::ActiveQuery` and checking if `loaded?` is working as expected.

### CODE REVIEW

**Comment by josemarialopez** [comment]
[Implement the 'loaded?' functionality for the queries at ActiveForce gem](https://trello.com/c/coRJn8UO/1478-implement-the-loaded-functionality-for-the-queries-at-activeforce-gem)

**Review by JeffLuckett — APPROVED** [review]
Looks great.  The more AR compliant features, the better!

File: spec/active_force/active_query_spec.rb [hunk 1]
```diff
@@ -315,4 +315,24 @@
...
+    subject { active_query.loaded? }
+
+    before do
+      active_query.instance_variable_set(:@records, records)
+    end
+
+    context 'when there are records loaded in memory' do
+      let(:records) { nil }
+
+      it { is_expected.to be_falsey }
```
[the remark this answers is not in the document]
Comment (josemarialopez) [reply spec/active_force/active_query_spec.rb:329]: @hstrowd , yes, you are absolutely right. I was having that idea also in mind but somehow I changed the functionality at the end. Updating now..

[the remark this answers is not in the document]
Comment (josemarialopez) [reply spec/active_force/active_query_spec.rb:329]: Changes applied! 🚀

**Review by hstrowd — APPROVED** [review]
Looks great. 👍

**Pull request merged**